### PR TITLE
PT-1410 Password reset: distinguish information message and reset message containing key for patient

### PR DESCRIPTION
**Pull request by franck-fourel**
- the patient will get an information message with instructions when asking for a password reset from the web app 

- the patient will get the actual key to make the reset when asking for a password reset from the handset

### CODE REVIEW

**Review by BigFab — COMMENTED** [review]
HTML found in `source` are supposed to have CSS classes declared but no inline CSS.
HTML found in `templates` have the CSS inlined.

File: api/forgot.go [hunk 1]
```diff
@@ -96,9 +107,8 @@ func (a *Api) passwordReset(res http.ResponseWriter, req *http.Request, vars map
 		resetCnf.UpdateStatus(models.StatusCompleted)
 	}
 
-	if resetCnf != nil && a.addOrUpdateConfirmation(resetCnf, res) {
+	if resetCnf != nil && (info != nil || a.addOrUpdateConfirmation(resetCnf, res)) {
```
Comment (BigFab) [inline api/forgot.go:110]: so this means we do not consider the failure of storing the confirmation in the DB when we deal with an info email? I do not understand why

File: models/confirmation.go [hunk 2]
```diff
@@ -88,11 +89,17 @@ var (
 func NewConfirmation(theType Type, templateName TemplateName, creatorId string) (*Confirmation, error) {
 
 	shortKey := ""
+	status := StatusPending
 	var err error = nil
-	if theType == TypePatientPasswordReset {
+
+	switch theType {
```
Comment (BigFab) [inline models/confirmation.go:95]: I personally find it weird to have this logic specific to one template type to be added in the generic constructor. I would rather have pushed that in the "forgot.go"

File: models/confirmation_test.go [hunk 3]
```diff
@@ -147,6 +147,59 @@ func Test_NewPatientPasswordResetConfirmation(t *testing.T) {
 
 }
 
+func Test_NewPatientPasswordInfoConfirmation(t *testing.T) {
```
Comment (BigFab) [inline models/confirmation_test.go:225]: that makes me think I did not add one for the Type PatientPinReset. I will create a task for doing that later

**Review by BigFab — APPROVED** [review]
LGTM

File: api/forgot.go (hunk 1 above)
Comment (franck-fourel) [reply api/forgot.go:110]: This one is used to save confirmation if info is nil. In the case info is not nil the addOrUpdateConfirmation will not be called and so the confirmation will not be saved in the database.

Comment (BigFab) [reply api/forgot.go:110]: ok got it

**Pull request merged**